cogs/moderation.py

```diff
@@ -73,15 +77,16 @@ async def clear(self, ctx, amount: int):
         else:
             await ctx.send('please specify the number of messages to clear')
 
-    @commands.command()
-    @commands.bot_has_permissions(manage_roles=True)
-    @commands.has_permissions(manage_roles=True, manage_messages=True)
-    async def promote(self, ctx, member: discord.Member, role: discord.Role):
+    @command()
+    @bot_has_permissions(manage_roles=True)
+    @has_permissions(manage_roles=True, manage_messages=True)
+    async def promote(self, ctx: Context, member: discord.Member, role: discord.Role) -> None:
         """Promote member to role."""
+        # TODO: A custom check can handle this
         if role >= ctx.author.top_role:
             await ctx.send(embed=failure("Role needs to be below you in hierarchy."))
             return
-        elif role in member.roles:
+        if role in member.roles:
```

Comment:
Why use if?

Reply:
This part of code will always be called anyway, in case the first if will pass, it will skip to `return`, if not, it will go to `elif`. We can use `if` as `elif` doesn't provide any advantage there.